### DISCUSSION TITLE
Getting arm working

### DISCUSSION THREAD

**Discussion by ryjones** [pq-code-package/tsc#66] [General]
@hanno-becker [Could you help me out?](https://github.com/pq-code-package/mlkem-c-aarch64/tree/GHA)

It looks like a lot of this isn't available for ARM.

**Comment by cothan** [comment]
So I spent exactly 30 minutes at this problem, I could be wrong here. 

This series of command failed: `brew install litani cbmc cbmc-viewer` because there is no brew. 

The hiccup is CBMC, it's not available in Nix (https://mynixos.com/search?q=homebrew), except Nix-darwin, which is Nix on MacOS. 
I tried to install CBMC on my `pi5`, but the `cbmc` package is not available, although it's available [here](https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=cbmc)
No `Litani` and `cbmc-viewer` on Debian and Ubuntu packages. 
The Linux road for CBMC seem like we need to compile from source, no `apt-get` at least. 

Or we can ask Linux to install brew. 

To install CBMC on Linux, one can install brew, but homebrew is not available on Linux. 
Other approach is to build from source for `litani, cbmc, cbmc-viewer`, but this would significantly increase the CI time. (Or we can build it somewhere then fetch it). 

The easiest approach is to make the `pqcp-arm64` CI to be macOS. 

Otherwise we have to engineer our build system somehow to have brew and hope brew make the rest working.

**Reply by ryjones** [reply]
when I did this, it said brew on linux arm64 is not supported.

**Comment by hanno-becker** [comment]
@ryjones Thank you for setting this up!

We don't have to run CBMC on all CI runners. If it's the only issue, let's remove it from this one. I'm having a look.

**Comment by hanno-becker** [comment]
@ryjones Things work fine without CBMC (https://github.com/pq-code-package/mlkem-c-aarch64/actions/runs/9394675772/job/25872762361), so let's get the `pqcp-arm64` runner going with build + test to begin with. I'll open a PR. Thank you again 🙏
